### DISCUSSION TITLE
PyTorch Lightning for New Examples

### DISCUSSION THREAD

**Discussion by RaivoKoot** [pykale/pykale#43] [Ideas] [answered]
This is based on the following issue: `https://github.com/pykale/pykale/issues/42` #42 , where I propose that for any future examples where we train models, we should use pytorch lightning.  
Haiping pointed out 
```
One issue is about non-deep learning flows. We need to find out

whether they can be fit in Pytorch Lightning (PL) well,
whether a PL implementation will be beneficial or detrimental, and
if PL does not fit, whether we can sort out something similar, as a key feature of PyKale.
```

## Response @haipinglu 
I think at least all deep learning examples should use PL. For non-deep learning flows, I am unsure. I think it would be a big improvement already to simply start out by having no rule for non-deep learning flows ( as we do it right now, so no change) and requiring PL for any deep learning flows.

**Comment by haipinglu** [comment]
@pz-white Is this feasible for GNN/GCN modules? @
@XianyuanLiu I assume this is not an issue for you, right? Anything to add?
@sz144 How can we standardise non-DL modules with our current API design and learning from PL? Is this feasible/necessary/beneficial?

**Reply by xianyuanliu** [reply]
No problem for me. Action examples will all use PL. I haven't found any bugs or issues with PL in my work yet. If any, I will update.

**Reply by peizhenbai** [reply]
There might be some concerns for all GraphML examples using PL especially on model construct part, since most SOTA GCN based models are built with Pytorch Geometric, that's also why I didn't use PL for examples before. But I think it's a good way for us to unifying deep learning models with PL. PL can help to better mange parts except model construct. I'll try to convert PL in the following case build. Thanks!

**Reply by shuo-zhou** [reply]
It will be good and friendly for general users if we can standardise all modules (with PL) in Pykale. I think we will switch to PL finally. We may have a discussion whether it is something urgent in our to do list in the next meeting.

**Comment by xianyuanliu** [comment, marked as answer]
Deep-learning examples utilize PyTorch-Lightning, whereas non-deep-learning ones utilize at least PyTorch.
